### PR TITLE
Fix non-working zwave log andriod PWA

### LINKED ISSUE #1681: Zwave pop up log does not work
**Home Assistant release with the issue:**
0.78

**Last working Home Assistant release (if known):**
0.77.3

**Browser and Operating System:**
Progressive web app (using Chrome) on Android 8.1.

**Description of problem:**
Pop up window does not show log when "Load" or "Tail" are clicked.

See below videos:

Expected behaviour:
https://imgur.com/gallery/cz8xGwC

Behaviour in Progressive web app (pop up is blank):
https://imgur.com/gallery/DgXrfn6

### PR DESCRIPTION
**Pull request by turbokongen**
This fixes #1681 
Workaround to keep all features intact is to add dialog to PWA(standalone mode) and use window in browser mode.

### CODE REVIEW

File: src/panels/config/zwave/zwave-log.js [hunk 1]
```diff
@@ -39,6 +41,12 @@ class OzwLog extends PolymerElement {
           <paper-button raised="true" on-click="_openLogWindow">Load</paper-button>   
           <paper-button raised="true" on-click="_tailLog" disabled="{{_completeLog}}">Tail</paper-button>
       </paper-card>
+      <paper-dialog id="pwaDialog">
```
Comment (balloob) [inline src/panels/config/zwave/zwave-log.js]: This is not how we do dialogs. You need to register a dialog.

File: src/panels/config/zwave/zwave-log.js [hunk 2]
```diff
@@ -100,5 +114,9 @@ class OzwLog extends PolymerElement {
       this.setProperties({ _completeLog: false });
     } else { this.setProperties({ _completeLog: true }); }
   }
+
+  _isPwa() {
```
Comment (balloob) [inline src/panels/config/zwave/zwave-log.js]: Please make this a standalone function in `common/`

**Comment by turbokongen** [comment]
I need some advice:
I'm stuck at getting a callback from the dialog to know if it closed or not.
I tried some stuff I found in the repo, but I can't seem to get it to work.
Everything works as expected, but I can't get the closed event back into the main log page.
Any hint would be helpful at this point 🥇

File: src/panels/config/zwave/zwave-log.js [hunk 3]
```diff
@@ -97,13 +99,16 @@ class OzwLog extends PolymerElement {
   }
 
   async _refreshLog(ozwWindow) {
-    if (ozwWindow.closed === true || (isPwa() && this.$.pwaDialog.opened === false)) {
+    if (ozwWindow.closed === true || (isPwa() && this._pwaDialogClosed === true)) {
       clearInterval(this._intervalId);
+      console.log('Clearing');
```
Comment (balloob) [inline src/panels/config/zwave/zwave-log.js]: extra log

File: src/panels/config/zwave/zwave-log-dialog.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    console.log('showDialog');
+    this._ozwLog = _ozwLog;
+    this._opened = true;
+    this._dialogClosedCallback = dialogClosedCallback;
+    setTimeout(() => this.$.pwaDialog.center(), 0);
+  }
+
+  _dialogClosed(ev) {
+    console.log('_dialogClosed', ev.target.nodeName);
+    if (ev.target.nodeId === 'ZWAVE-LOG-DIALOG') {
```
Comment (balloob) [inline src/panels/config/zwave/zwave-log-dialog.js]: What's with all this logic?

File: src/panels/config/zwave/zwave-log.js [hunk 5]
```diff
@@ -100,5 +120,31 @@ class OzwLog extends PolymerElement {
...
+        dialogTag: 'zwave-log-dialog',
+        dialogImport: () => import('./zwave-log-dialog.js'),
+      });
+    }
+  }
+
+  _showOzwlogDialog() {
+    console.log('_showOzwlogDialog');
+    this.fire('show-ozwlog-dialog', {
+      _ozwLog: this._ozwLogs,
```
Comment (balloob) [inline src/panels/config/zwave/zwave-log.js]: I don't see any logic to keep this in sync when the dialog is open?

**Pull request merged**